src/FSharpLanguageServer/ProjectManager.fs

```diff
@@ -336,8 +337,34 @@ type ProjectManager(checker: FSharpChecker) =
         walk(root.resolved.Value.options)
         List.ofSeq(result)
 
-    member this.AddWorkspaceRoot(root: DirectoryInfo): Async<unit> = 
+    /// Normalizes directory separator before comparing
+    let equalPaths (p1: string) (p2: string) =
+        p1.Replace("\\", "/") = p2.Replace("\\", "/")
+
+    let getSolutionProjectFiles (slnPath: FileInfo) =
+        let reg = System.Text.RegularExpressions.Regex("""^Project\(".*?"\) = ".*?", "(.*)",""")
+        File.ReadLines slnPath.FullName
+        |> Seq.choose (fun line ->
+            let m = reg.Match(line)
+            if m.Success then
+                let relativePath = m.Groups.[1].Value
+                if relativePath.EndsWith(".fsproj") then
+                    Path.Combine(slnPath.Directory.FullName, relativePath)
+                    |> Path.GetFullPath |> Some
+                else None
+            else None)
+        |> Seq.toList
+
+    member this.AddWorkspaceRoot(root: DirectoryInfo): Async<unit> =         
         async {
+            solutionProjects <-
+                root.EnumerateFiles("*.sln", SearchOption.TopDirectoryOnly)
+                |> Seq.tryHead
+                |> Option.map (fun slnPath ->
+                    let slnProjs = getSolutionProjectFiles slnPath
+                    dprintfn "Found solution %s with .fsproj references: %A" slnPath.FullName slnProjs
+                    slnProjs)
+                |> Option.defaultValue []
```

Comment:
I just realized that this won't work with multiple workspaces. I've never used more than one workspace in a single VS Code instance, so I'm not sure what's the best way to deal with that case.